addOns/ascanrules/src/test/java/org/zaproxy/zap/extension/ascanrules/CrossSiteScriptingScanRuleUnitTest.java

```diff
@@ -961,6 +961,40 @@ protected Response serve(IHTTPSession session) {
         assertThat(alertsRaised.get(0).getConfidence(), equalTo(Alert.CONFIDENCE_MEDIUM));
     }
 
+    @Test
+    void shouldNotReportXssInUrlAttributeWhenPayloadIsModified()
+            throws NullPointerException, IOException {
+        String test = "/shouldNotReportXssInUrlAttributeWhenPayloadIsModified/";
+
+        this.nano.addHandler(
+                new NanoServerHandler(test) {
+                    @Override
+                    protected Response serve(IHTTPSession session) {
+                        String name = getFirstParamValue(session, "name");
+                        String response;
+                        if (name != null) {
+                            // Strip out < and >
+                            name = name.replaceAll("<", "").replaceAll(">", "");
```

Comment:
Didn't also replace `"` on purpose?

Reply:
Do you mean replace `"` with an empty string in the attack payload? Like this:
`name = name.replaceAll("<", "").replaceAll(">", "").replaceAll('"', '');`

Reply:
yes